app/Services/AirfieldService.php

```diff
@@ -2,28 +2,34 @@
 
 namespace App\Services;
 
-use App\Models\Airfield;
+use App\Models\Airfield\Airfield;
 
 class AirfieldService
 {
     /**
      * @return array
      */
-    public function getAllAirfieldsWithTopDown() : array
+    public function getAllAirfieldsWithRelations() : array
     {
-        $airfields = Airfield::all();
+        $airfields = [];
+        Airfield::all()->each(function (Airfield $airfield) use (&$airfields) {
+            $prenotePairings = $airfield->prenotePairings()->select(['destination_airfield_id', 'prenote_id'])->get();
+            $prenoteArray = [];
 
-        $airfieldArray = [];
-        $airfields->each(function (Airfield $airfield) use (&$airfieldArray) {
-            $airfieldArray[] = array_merge(
+            $prenotePairings->each(function (Airfield $airfield) use (&$prenoteArray) {
```

Comment:
Another candidate for `mapWithKeys`?

Reply:
Don't think we can on this one as it's adding to arrays as it goes :)